src/httprpc.cpp

```diff
@@ -72,6 +75,58 @@ static void JSONErrorReply(HTTPRequest* req, const UniValue& objError, const Uni
     req->WriteReply(nStatus, strReply);
 }
 
+//This function checks username and password against -rpcauth
+//entries from config file.
+static bool multiUserAuthorized(std::string strUserPass)
+{    
+    std::string strUser = strUserPass.substr(0, strUserPass.find(":"));
+    std::string strPass = strUserPass.substr(strUserPass.find(":") + 1);
+
+    if (mapMultiArgs.count("-rpcauth") > 0) {
+        //Search for multi-user login/pass "rpcauth" from config
+        BOOST_FOREACH(std::string strRPCAuth, mapMultiArgs["-rpcauth"])
+        {
+            std::vector<std::string> vFields;
+            boost::split(vFields, strRPCAuth, boost::is_any_of(":$"));
+            if (vFields.size() != 4) {
+                //Incorrect formatting in config file
+                continue;
+            }
+
+            std::string strName = vFields[0];
+            if (!TimingResistantEqual(strName, strUser)) {
+                continue;
+            }
+
+            std::string strIterations = vFields[1];
+            int nIterint = atoi64(strIterations);
+            std::string strSalt = vFields[2];
+            std::string strHash = vFields[3];
+
+            unsigned int KEY_SIZE = 32;
+            unsigned int KEY_HEX_SIZE = 64;
+            unsigned char *out = new unsigned char[KEY_SIZE]; 
+            char outhex[KEY_HEX_SIZE];
+
+            if (PKCS5_PBKDF2_HMAC(strPass.c_str(), strPass.size(), 
```

Comment:
I think we really want to avoid adding OpenSSL dependencies. But it shouldn't be hard to add PBKDF2 (we already have a builtin HMAC-SHA256 implementation).